---
title: "Multi-Omics Sleep Disorder Breathing"
author: "Nuzulul Kurniansyah & Tamar Sofer"
date: "12/29/2021"
output: 
  md_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction


This repository provides information regarding transcriptome wide-association test and the construction of a transcript polygenic risk score (tPRS) for sleep disorder breathing (SDB) that we developed in manuscript "An integrated multi-omics analysis of sleep-disordered breathing traits across multiple blood cell types" (link to be added).

First, it provides instructions for constructing the tPRS based on transcript GWAS. We provide the relevant for each transcript summary statistics (see folder “Summary_Statistics_for_tPRS_construction”), as well as code for using them to construct the tPRS.

Second, this repository also provides code the we used for the analyses in the manuscript (see folder “Code”).


## Transcriptome wide-association test
We performed Transcript wide association analysis of SDB phenotype (AvgO2, MinO2, AHI, and multiple SDB traits) in multiple blood tissue in MESA (Multi-Ethnic Study of Atherosclerosis) using [Olivia](https://github.com/nkurniansyah/Olivia "Olivia") R package that we developed our previous works in manuscript [Benchmarking association analyses of continuous exposures with RNA-seq in observational studies](https://academic.oup.com/bib/article-abstract/22/6/bbab194/6278609).

Briefly, we performed median normalization for gene-level expected counts and then filtered lowly expressed gene transcripts defined by removing transcripts with the proportion of zero higher than 0.5, median value lower than 1, maximum expression range value lower than 5, and maximum expression value lower than 10. Transcript counts were log-transformed after counts of zero were replaced with half the minimum of the observed transcript count in the sample. The analyses were adjusted for age, sex, study center, race/ethnic group, and batch variables: plates, shipment batch, and study site. Because BMI is a strong risk factor for SDB and is assumed to be part of the causal chain, we conducted additional analyses adjusting for BMI (BMIadj). We computed empirical p-values to account for the highly skewed distribution of SDB phenotypes implemented in Olivia R package, which may lead to false-negative associations if ignored. Finally, we accounted for multiple testing by applying False-Discovery Rate (FDR) correction to each association analysis using the Benjamini-Hochberg (BH) procedure. We carried forward transcript associations with FDR p-value<0.1 


## Transcript genome-wide association study (GWAS)


The next step is to perform a Genome-wide Association Study for each transcript with FDR p-value < 0.1 (96 transcript unadjusted BMI and 24 transcriptadjusted BMI analysis) using the MESA reference panel. We followed the guideline to perform GWAS using [TopmedPipeline](https://github.com/UW-GAC/analysis_pipeline "TopmedPipeline").


## Constructing tPRS

After performing GWAS, we defined SNPs selection using cis-QTLs limited to the transcript coding region, and then we used [PRSice 2.3.1.e](https://www.prsice.info "PRSice 2.3.1.e") to construct tPRS for each transcript GWAS. This command below is to construct tPRS using the transcript GWAS we provide (./Summary_Statistics_for_tPRS_construction). The summary statistics are based on the specific set of SNPs selected after the validation step in the independent data set(see manuscript for details). Note that genetic data files must be specified in the –target argument.


```{bash eval=FALSE, echo=TRUE}


Rscript ./PRSice.R \
 --dir ./PRS_Output \
 --prsice ./PRSice_linux/PRSice_linux \
 --base ./Summary_Statistics_for_tPRS_construction/. \
 --target ./Genotype \
 --thread 2 \
 --chr Chromosome 
 --bp Position 
 --A1 Allele1 
 --A2 Allele2 
 --pvalue PValue \
 --bar-levels Threshold \
 --stat BETA 
 --all-score T \
 --out ./out_prs \
 --no-clump T
 --print-snp T \
 --ignore-fid T 
 --no-regress T 
 --fastscore T 
 --model add 
 --no-full T 
 --chr-id c:l:a:b


```


## Example code for association analsis

We performed association analysis for each tPRS using mixed models implemented in the GENESIS R package. Below is an example code. It uses function that we provide in the folder "Code".

```{r eval=FALSE, echo=TRUE}

library(GENESIS)
library(GWASTools)
library(data.table)

source("./Code/*")


#phenotype

pheno<- fread(phenotype_file, data.table=F)


# merge tPRS with phenotype

tPRS_transcript1<- fread(tPRS_transcript1_file, data.table=F)

tPRS_transcript1[,"tPRS"]<- sapply(tPRS_transcript1[,"tPRS"], function (x) (x-mean(x))/sd(x))


pheno_df<-left_join(pheno,tPRS_transcript1, by="sample.id" )


covarites_prs<- c("age","sex","site","race",paste0("PC_",1:5),"tPRS")

#example outcome
outcome<-"AvgO2"

## Kinship matrix

covMatlist<-getobj(covMatlist)


assoc_df<- run_assoc_mixmodel(pheno=pheno,
                              outcome=outcome,
                              covars_prs=covarites_prs, 
                              covmat=covMatlist,
                              group.var=NULL,
                              inverse_normal=TRUE)


```




## Example code for metebolome wide-association test

We performed  metebolome wide-association test for each tPRS with metabolomics data. see example below:



```{r eval=FALSE, echo=TRUE}


library(data.table)
library(survey)
library(dplyr)
library(GWASTools)
library(tidyverse)

source("./Code/*")


#phenotype

pheno<- fread(phenotype_file, data.table=F)


# merge tPRS with phenotype

tPRS_transcript1<- fread(tPRS_transcript1_file, data.table=F)

tPRS_transcript1[,"tPRS"]<- sapply(tPRS_transcript1[,"tPRS"], function (x) (x-mean(x))/sd(x))


pheno_df<-left_join(pheno,tPRS_transcript1, by="sample.id" )


covarites<- c("age","sex","site","race",paste0("PC_",1:5))


metab<- getobj(metab_file)

IDs_both <- intersect(pheno_df$sample.id, rownames(metab))

rnaseq_matrix <- rnaseq_count_matrix[IDs_both, ] 

metab <- pheno_df[match(IDs_both,(pheno_df$sample.id)),]

# Drop samples with more than ...% missing (25%)
drop_sample_ind<-which(rowSums(is.na(metab)) < (ncol(metab)-2)* 0.25)
metab<-metab[drop_sample_ind,]


# Drop metabolites with more than ...% missing (75 %)
drop_metab_ind <- which(colSums(is.na(metab)) < nrow(metab)*0.75)
metab<-metab[,drop_metab_ind]

# Imputed missing Metab with min value
metab_imp<-data.frame(sapply(metab,function(x) ifelse(is.na(x),min(x, na.rm = TRUE),x)))

### rank normalize metab
metab_imp<-sapply(metab_imp, as.numeric)
metab_imp<-sapply(metab_imp, rank_normalization)

metab_imp<-data.frame(metab_imp)%>% rownames_to_column(var="sample.id")


complete_pheno<-left_join(pheno_df,metab_imp, by="sample.id")

list_metab<-colnames(metab_imp)[2:ncol(metab_imp)]



metab_assoc<-run_metab_assoc(phenotype = complete_pheno,
                             sample_weight = "WEIGHT_FINAL_NORM_OVERALL",
                             psu_id = "PSU_ID",
                             strata = "STRAT",
                             covars = covarites,
                             exposure = "tPRS",
                             metab_include =list_metab )



metab_assoc<-data.frame(metab_assoc)
row.names(metab_assoc)<-NULL
# remove unknow metabolite
identify_metab<-metab_assoc%>% dplyr::filter(!str_detect(Metabolite,"^X"))

## Add FDR

assoc_clean_df<- identify_metab %>% mutate(FDR_BH=p.adjust(P.value,method="BH"))


metab_output<-args[5]
write.csv(assoc_clean_df, file=metab_output, row.names = F)

```



